### PR TITLE
Makefile to run tests

### PR DESCRIPTION
**Pull request by pdietl**

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -0,0 +1,32 @@
...
+all: $(OUTPUT)
+
+$(OUTPUT): makeself.sh makeself-header.sh VERSION
+	./make-release.sh
+
+clean:
+	$(RM) makeself-*.run
+
+test:
+	cd test && git submodule update --init --recursive
```
Comment (realtime-neil) [inline Makefile]: You would test makeself using the latest `HEAD` of bashunit at the time `make test` is invoked? I disagree with the opinion that informed this decision on the grounds that builds --- and the tests thereon --- should be as repeatable as possible. Furthermore, I think it's a mistake for the build and/or test to require anything more than the source code --- the presence of a git working directory should be optional.

File: Makefile [hunk 2]
```diff
@@ -0,0 +1,32 @@
...
+	$(foreach f, \
+		$(notdir $(filter-out test/bashunit,$(wildcard test/*))), \
```
Comment (realtime-neil) [inline Makefile]: Here, by not sorting the output of the `wildcard` operator, one allows a nondeterministic ordering of tests. I personally prefer an explicit enumeration of test executables, but if you must use globbing expressions, please sort the result.

File: Makefile (hunk 1 above)
Comment (pdietl) [reply Makefile]: I agree with you, Unfortunately bashunit has not made any tags or releases, so I have requested this here https://github.com/djui/bashunit/issues/7

Comment (pdietl) [reply Makefile]: Actually, if you do a git log, the bashunit submodule is frozen to whenever it was added, some commit in 2014, I think.

Comment (realtime-neil) [reply Makefile]: Meh, it's enough that makeself source control knows which bashunit sha1 hash it wants. The presence of tags (or even branches) within the bash unit remote is hardly necessary for projects that wish to consume it.

Comment (pdietl) [reply Makefile]: indeed

Comment (realtime-neil) [reply Makefile]: > Actually, if you do a git log, the bashunit submodule is frozen to whenever it was added, some commit in 2014, I think.

Granted, and I agree that the reference should be updated, but as part of a different feature request.

**Comment by pdietl** [comment]
@realtime-neil any other changes I should make?

**Comment by realtime-neil** [comment]
> @realtime-neil any other changes I should make?

Lose the `git submodule update ...` and you've got my vote.

**Comment by realtime-neil** [comment]
Beautiful; thanks much @pdietl .

**Pull request merged**